src/components/app.tsx

```diff
@@ -0,0 +1,77 @@
+import { State } from 'ketting';
+import React from 'react';
+import url from 'url';
+import { PageProps } from '../types';
+
+import { Search } from './search';
+import { Navigation } from './navigation';
+import { Alternate } from './alternate';
+import { Resource } from './resource';
+
+export function App(props: PageProps) {
+
+  const stylesheets = props.options.stylesheets.map( ss => {
+    return <link rel="stylesheet" href={url.resolve(props.options.assetBaseUrl, ss)} type="text/css" />;
+  });
+
+  const resourceTitle = getResourceTitle(props.resourceState);
```

Comment:
Could something like this go inside a `useEffect` to manage rendering if the `props.resourceState` is a changing property?

Reply:
I could be wrong but in the server-side everything is basically synchronous. useEffect implies a re-render, and at that time it's already 'too late' because we sent the data to the HTTP client.